src/Blocks/BlockComparator.h

```diff
@@ -43,9 +43,32 @@ class cBlockComparatorHandler :
 		return true;
 	}
 
-	virtual bool CanBeAt(cChunkInterface & a_ChunkInterface, int a_RelX, int a_RelY, int a_RelZ, const cChunk & a_Chunk) override
+	virtual bool CanBeAt(cChunkInterface & a_ChunkInterface, int a_RelX, int a_RelY, int a_RelZ, const cChunk & a_Chunk, NIBBLETYPE a_BlockMeta) override
 	{
-		return ((a_RelY > 0) && (a_Chunk.GetBlock(a_RelX, a_RelY - 1, a_RelZ) != E_BLOCK_AIR));
+		if (a_RelY <= 0)
+		{
+			return false;
+		}
+
+		BLOCKTYPE BlockIsOnType;
+		NIBBLETYPE BlockIsOnMeta;
+		a_Chunk.UnboundedRelGetBlock(a_RelX, a_RelY - 1, a_RelZ, BlockIsOnType, BlockIsOnMeta);
+
+		// On the top of an upside-down slab
+		if (cBlockSlabHandler::IsAnySlabType(BlockIsOnType))
+		{
+			// Check if the slab is turned up side down
+			return (cBlockSlabHandler::IsUpsideDown(BlockIsOnMeta));
+		}
+
+		// On the top of an upside-down stairs
+		if (cBlockStairsHandler::IsAnyStairType(BlockIsOnType))
+		{
+			return (cBlockStairsHandler::IsUpsideDown(BlockIsOnMeta));
+		}
+
+		bool Result = cBlockInfo::IsFullSolidOpaqueBlock(BlockIsOnType);
+		return Result;
```

Comment:
comparator can not put on TNT is not mentioned in wiki, and I tested in venilla, I can't put in on TNT too, I will add more check about TNT in this method

Reply:
I think this indicates that TNT isn't `FullSolidOpaque`, there were a few items I tested that shouldn't have been placable on TNT.

Reply:
I noticed in TNT wiki:
https://minecraft.gamepedia.com/TNT

> Transparency : Partial

So I think TNT block property `m_Transparent` should set to true to avoid this issue?

Reply:
I think that would break the lighting calculations.  Maybe just add a special case to `IsFullSolidOpaque`?

Reply:
hi @peterbell10 ,

I tested after add TNT special case in `IsFullSolidOpaque`, and compared with vanilla server, there are some behavior confict with vanilla server.

1. torch can be put on the side of TNT in vanilla, and it can not in cuberite.
2. button can be put onto TNT in vanilla, and can not in cuberite.

I think it is better to let TNT as a normal `full solid opaque` block and add special case in some block handle.

Reply:
In that case redstone and repeates will also need special cases.